include/sys/euiAnimator.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <nn/font/font_Util.h>
+#include <nn/ui2d/ui2d_Animation.h>
+
+namespace nn {
+namespace ui2d {
+class Group;
+class Pane;
+}  // namespace ui2d
+}  // namespace nn
+
+namespace eui {
+class LayoutEx;
+
+class Animator : public nn::ui2d::AnimTransformBasic {
+public:
+    NN_RUNTIME_TYPEINFO(AnimTransformBasic);
+
+    enum PlayType { PLAYTYPE_ONESHOT, PLAYTYPE_LOOP, PLAYTYPE_ROUNDTRIP };

```

Comment:
Should be `enum class` to avoid it being replaced by the base type in function symbols.

Take this with a grain of salt, I don't really know if the above statement is true - `enum class` is preferred by the contribution guidelines either way, if there is no indication that this should be a normal `enum` instead of `enum class`.

Reply:
I'm 100% sure that `enum class` was not used at all.

Reply:
How do you know that / how would you tell apart whether `enum class` or `enum` was used?

Reply:
`enum` seems to be the most likely implementation for this case. There isn't really a way to tell, I just simply used `enum` as it seemed like Nintendo.

I could change it to `enum class` if it really bugs you or any of the other maintainers.

Reply:
there are cases where enum is more likely (in this case maybe there was a print or string for these values), also I'm pretty sure enum still produces the proper symbol

Reply:
Unless we have debug info or asserts or whatever, I don't think it's possible to tell if they used a regular enum or an enum class. Both types of enums are mangled the exact same way.

In this case it's more idiomatic to use an enum class - we don't rely on implicit conversion to int and there is no good reason for the enumerators to leak into their surrounding scope. The PLAYTYPE_ prefix and SCREAMING would be unnecessary with an enum class.

In general what the original Nintendo code does is merely a guideline rather than something to be followed rigidly (as long as things produce the same codegen). We frequently deviate from what we think N did if it makes the code clearer or if we can modernise the code without introducing codegen differences.

Reply:
Alright, I'll change it to an `enum class` because I don't feel like arguing about this much more.